rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1106,6 +1123,41 @@ private void processReleaseRequests() {
         });
     }
 
+    private void processReleasesInBatch(ReleaseRequestQueue releaseRequestQueue,
+                                        ReleaseTransactionBuilder txBuilder,
+                                        List<UTXO> availableUTXOs,
+                                        ReleaseTransactionSet releaseTransactionSet) {
+        long currentBlockNumber = rskExecutionBlock.getNumber();
+        long nextPegoutCreationBlockNumber = getNextPegoutCreationBlockNumber();
+
+        if (currentBlockNumber >= nextPegoutCreationBlockNumber) {
+            // batch pegout transactions
+            Optional<ReleaseTransactionBuilder.BuildResult> result = txBuilder.buildBatchedPegouts(releaseRequestQueue.getEntries());
+
+            if (!result.isPresent()) {
+                logger.warn(
+                        "Couldn't build a release BTC tx for <{}, with sum {}>",
+                        releaseRequestQueue.getEntries().hashCode(),
+                        releaseRequestQueue.getEntries().stream().mapToDouble(e -> e.getAmount().value).sum());
+                return;
+            }
+
+            BtcTransaction generatedTransaction = result.get().getBtcTx();
+            addPegoutTxToReleaseTransactionSet(generatedTransaction, releaseTransactionSet);
+
+            // Mark UTXOs as spent
+            List<UTXO> selectedUTXOs = result.get().getSelectedUTXOs();
+            availableUTXOs.removeAll(selectedUTXOs);
+
+            // update next Pegout height
+            long nextPegoutHeight = currentBlockNumber + bridgeConstants.getNumberOfBlocksBetweenPegouts();
+            provider.setNextPegoutHeight(nextPegoutHeight);
+
+            adjustBalancesIfChangeOutputWasDust(generatedTransaction,
+                Coin.valueOf(releaseRequestQueue.getEntries().stream().mapToLong(e -> e.getAmount().value).sum()));
```

Comment:
```suggestion
            Coin totalValue = releaseRequestQueue.getEntries().stream().reduce(Coin.ZERO, (r1, r2) -> r1.getAmount().add(r2.getAmount()));
            adjustBalancesIfChangeOutputWasDust(generatedTransaction, totalValue);
```
Another option, not sure if it affects performance in any way

Reply:
Coin totalValue = releaseRequestQueue.getEntries().stream().map(ReleaseRequestQueue.Entry::getAmount).reduce(Coin.ZERO, Coin::add);